src/Mvc/Mvc.ViewFeatures/src/HtmlHelperComponentExtensions.cs

```diff
@@ -14,21 +14,21 @@ namespace Microsoft.AspNetCore.Mvc.Rendering
     /// <summary>
     /// Extensions for rendering components.
     /// </summary>
-    public static class HtmlHelperRazorComponentExtensions
+    public static class HtmlHelperComponentExtensions
     {
         /// <summary>
         /// Renders the <typeparamref name="TComponent"/> <see cref="IComponent"/>.
         /// </summary>
         /// <param name="htmlHelper">The <see cref="IHtmlHelper"/>.</param>
         /// <returns>The HTML produced by the rendered <typeparamref name="TComponent"/>.</returns>
-        public static Task<IHtmlContent> RenderStaticComponentAsync<TComponent>(this IHtmlHelper htmlHelper) where TComponent : IComponent
+        public static Task<IHtmlContent> RenderComponentAsync<TComponent>(this IHtmlHelper htmlHelper) where TComponent : IComponent
```

Comment:
Won't this cause issues once we reintroduce stateful pre-rendering?

Reply:
I think we'd try to avoid introducing another set of APIs. This is the name we want to use for now.